[docs] Missing annotations

hongkunyoo opened the pull request:
Not quite sure but It seems that `nginx.ingress.kubernetes.io/canary-by-header` is missing.

## What this PR does / why we need it:
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->
Just a little typo.

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation only

## Which issue/s this PR fixes
<!--
(optional, in `fixes #<issue number>` format, will close that issue when PR gets merged):

fixes #
-->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/kubernetes/ingress-nginx/blob/main/CONTRIBUTING.md) guide
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

kundan2707 commented:
@hongkunyoo i am not able to see if anything was missing. 
you have deleted same line and added same. can you check once again ?

hongkunyoo commented:
@kundan2707 
I have changed from
> This annotation has to be used together with .

to

> This annotation has to be used together with `nginx.ingress.kubernetes.io/canary-by-header`

I added `nginx.ingress.kubernetes.io/canary-by-header`.

kundan2707 commented:
@hongkunyoo  got it 
thanks

strongjz commented:
/lgtm
